elastic/eui
# [EuiButtonGroup] Convert to Emotion

## Summary

This PR converts `EuiButtonGroup` and `EuiButtonGroupButton` to Emotion. I recommend [following along by commit](https://github.com/elastic/eui/pull/6841/commits). TL;DR of changes:

- Adds a new internal `euiOutline()` CSS-in-JS util (a lighter weight version of `euiFocusRing()`, without the focus-visible styles)
- Removes the component's usage of `EuiButtonDisplayDeprecated` in favor of the new internal `EuiButtonDisplay` component, and then removes `EuiButtonDisplayDeprecated` entirely
- Fixes text truncation not working correctly on mobile/smaller screen widths (free fix w/ new `EuiButtonDisplay` usage)
- Fixes border separation between buttons not responding to light/dark mode
- Fixes focus ring not working as expected
  - Outlines now correctly _only_ appear on keyboard tab and not on mouse click
  - Outline colors on `compressed` button groups now correctly use the same color as the button fill
- Attempts to partially address https://github.com/elastic/eui/issues/6819 by increasing the font weight of selected options
- Cleans up a lot of unnecessary CSS cruft
- ✨ Adds a new Storybook page for `EuiButtonGroup`, making permutations significantly easier to test

## QA

- Compare [staging](https://eui.elastic.co/pr_6841/#/navigation/button#button-groups) to [production](https://elastic.github.io/eui/#/navigation/button#button-groups)
- [ ] Confirm the components look approximately the same as before without visual regression
- Note that the visuals are not _exactly_ the same - Emotion changes some slight rendering around font size and line height (now inherits directly from `EuiButtonDisplay`. I consider this to be an intentional change to match other existing buttons
- [OPTIONAL] View component in Storybook
    - `gh pr checkout 6841`
    - `yarn storybook`
    - Go to http://localhost:6006/?path=/story/euibuttongroup--playground and test multiple permutations of sizes, colors, etc 

### General checklist

~- [ ] Checked for **breaking changes** and labeled appropriately~ - Even though `EuiButtonDisplayDeprecated` was deleted, this was not a publicly exported component and therefore is not a breaking change

~- [ ] Props have proper **autodocs** (using `@default` if default values are missing) and **[playground toggles](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/playgrounds.md)**~
~- [ ] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting)**~
~- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples~
~- [ ] Updated the **[Figma](https://www.figma.com/community/file/964536385682658129)** library counterpart~

## Emotion checklist
### Acceptance criteria
~- [ ] All SCSS overrides have been removed from the Amsterdam theme~ - No overrides for this component
~- [ ] Any dependent components are identified in a new issue~

### Checklists

**Kibana usage**

  ~- [ ] If usage exists, consider converting to a `data` attribute so that consumers still have something to hook into~ - Retained `.euiButtonGroupButton-isSelected` className, which is the primary modifier usage

---

**General**


---

**Unit tests**

- [x] Converted from Enzyme to RTL
~- [ ] Removed any mounted snapshots~

---

**Sass/Emotion conversion process**


~- [ ] Deleted any `src/amsterdam/overrides/{component}.scss` files (styles within should have been converted to the baseline Emotion styles)~
~- [ ] Converted all global Sass vars/mixins to JS (e.g. `$euiSize` to `euiTheme.size.base`)~
~- [ ] Removed or converted component-specific Sass vars/mixins to exported JS versions~
~- [ ] Listed var/mixin removals in changelog~
~- [ ] Ran `yarn compile-scss` to update var/mixin [JSON files](https://github.com/elastic/eui/tree/main/src-docs/src/views/theme/_json)~
~- [ ] Simplified `calc()` to `mathWithUnits` if possible (if mixing different unit types, this may not be possible)~
~- [ ] Added an `@warn` deprecation message within the `global_styling/mixins/{component}.scss` file~

---

**CSS tech debt**

~- [ ] Used `gap` property to add margin between items if using flex~

---

**DOM Cleanup**


---

**Kibana due diligence**

~- [ ] Any test/query selectors that will need to be updated~ None
~- [ ] Any Sass or CSS that will need to be updated, particularly if a component Sass var was deleted~ None
~- [ ] Any direct className usages that will need to be refactored (e.g. someone calling the `euiBadge` class on a div instead of simply using the `EuiBadge` component)~ None

---

**Extras/nice-to-have**

~- [ ] Documentation pass~
~- [ ] Check for issues in the backlog that could be a quick fix for that component~
~- [ ] Optional component/code cleanup: consider splitting up the component into multiple children if it's overly verbose or difficult to reason about~
